### DISCUSSION TITLE
How to change primary key to use UUID

### DISCUSSION THREAD

**Discussion by ChicagoGregg** [directus/api#2445] [Q&A]
Hello, I'm running v8.8.1 with no collections, vanilla installation, and MySQL or MariaDB.

I saw #4441 and that didn't help me to resolve this and need to dig deeper. I'd like to switch the PK id to use UUID instead by default for all collections.

I saw this link: https://docs.directus.io/guides/collections/#

But I cannot see this option in Settings -> Data Model. I also tried to remove the old PK (well, and modify existing) and can't see any UUID options.

Are there any instructions to make this change? I'm starting from fresh with a new schema (so data loss isn't any issue).

Thanks in advance,

-gregg

**Comment by philleepflorence** [comment]
Hello, you would need to use the **v8.docs.directus.io** for 8.8.1:
- [Collections Guide](https://v8.docs.directus.io/guides/collections.html)
- [Field Data Types](https://v8.docs.directus.io/guides/field-types.html)
- [Primary Key Interface Guide](https://v8.docs.directus.io/guides/interfaces.html#primary-key)

If you have the requirements for D9, that has the option for [UUID](https://docs.directus.io/concepts/types/)

**Reply by ChicagoGregg** [reply]
Thank you @philleepflorence  for the reply and information.

I overlooked that the documentation is a different version (9) versus v8.

In reviewing the v8 documentation, it isn't clear anywhere on how to change it to use the uuid instead.

However, I went ahead and installed Directus v9 and it's very straight-forward with the UUID. I know it's still a RC, but I figure I'm developing the application right now anyways and it'll be that much closer when v9 officially comes out. :)

Thanks!
